File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        set<ListNode*> m;
        int index = 0;
        while (head)
        {
            if (m.insert(head).second == false)
                return (true);
            head = head->next;
            index++;       
        }
        return (false);
    }
};

File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        hashMap = [];
        while head:
            if (head in hashMap):
                return (True);
            hashMap.append(head);
            head = head.next;
        return (False);
        